Services/Common/OJS.Services.Common/Implementations/ContestsActivityService.cs

```diff
@@ -66,30 +74,41 @@ public bool CanBeCompeted(IContestForActivityServiceModel contest)
             return false;
         }
 
-        var now = this.dates.GetUtcNow();
+        var currentTimeInUtc = this.dates.GetUtcNow();
 
         if (!contest.EndTime.HasValue)
         {
             // Compete forever
-            return contest.StartTime <= now;
+            return contest.StartTime <= currentTimeInUtc;
         }
 
-        return contest.StartTime <= now && now <= contest.EndTime;
+        // If the above conditional statements are not entered, first we have to check the start and end time -
+        // if StartTime is before current moment and EndTime is after current moment contest CanBeCompeted.
+        // If this check returns false we have to check if the current user is a participant with remaining time
+        // in an online contest
+        return (contest.StartTime <= currentTimeInUtc && currentTimeInUtc <= contest.EndTime) ||
+               (contest.IsOnline && this.IsActiveParticipantInOnlineContest(contest.Id));
```

Comment:
I want to clarify something here. This will only be valid for the current user right?
There is another `IsActive` method that determines if the contest `CanBeCompeted` by **someone** whereas this method determines if the contest `CanBeCompeted` by **the current user**.

Reply:
IsActive checks if the contest has any participants with participation time left- needed property for different validations of contests. 
IsActiveParticipantInOnlineContest check if the **current** user is a participant in this contest with any participation time left which is needed to keep the contest's CanBeCompeted = true in order for the user to be able to open the contest and continue competing.